sdk/search/Azure.Search/src/Serialization/SearchDoubleConverter.cs

```diff
@@ -0,0 +1,72 @@
+﻿// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+using System;
+using System.Diagnostics;
+using System.Text.Json;
+using System.Text.Json.Serialization;
+using Azure.Core;
+
+namespace Azure.Search
+{
+    /// <summary>
+    /// Convert doubles to and from JSON.  Search allows INF, -INF, and NaN as
+    /// string values.
+    /// </summary>
+    internal class SearchDoubleConverter : JsonConverter<double>
+    {
+        public static SearchDoubleConverter Shared { get; } =
+            new SearchDoubleConverter();
+
+        public override double Read(
+            ref Utf8JsonReader reader,
+            Type typeToConvert,
+            JsonSerializerOptions options)
+        {
+            Debug.Assert(typeToConvert != null);
+            Debug.Assert(options != null);
+
+            if (reader.TokenType == JsonTokenType.String)
+            {
+                switch (reader.GetString())
+                {
+                    case Constants.InfValue:
+                        return double.PositiveInfinity;
```

Comment:
I am curious to learn about why this special handling is required. Is this defined in the Search rest api somewhere? Can you share a link if you have it. Thanks.

Reply:
The service implements an OData-compliant REST API and uses the Entity Data Model representations for data types. The representations for positive and negative infinity for type `Edm.Double` are "INF" and "-INF", respectively.